### PR TITLE
(METAMODEL-1226) Metadata for numeric data types does not include num…

### PR DESCRIPTION
**Pull request by tkingsbury42**
https://issues.apache.org/jira/browse/METAMODEL-1226

### CODE REVIEW

File: core/src/main/java/org/apache/metamodel/schema/Column.java [hunk 1]
```diff
@@ -85,6 +85,14 @@
      */
     public Integer getColumnSize();
 
+    /**
+     * Gets the number of decimal digits this column.
```
Comment (kaspersorensen) [inline core/src/main/java/org/apache/metamodel/schema/Column.java:89]: Nit-pick: Missing a "for" (this column) in this sentence.

File: core/src/main/java/org/apache/metamodel/schema/ImmutableColumn.java [hunk 2]
```diff
@@ -144,6 +147,9 @@ public Integer getColumnSize() {
         return columnSize;
     }
 
+    @Override
+    public Integer getDecimalDigits() { return decimalDigits; }
```
Comment (kaspersorensen) [inline core/src/main/java/org/apache/metamodel/schema/ImmutableColumn.java:151]: Styling-wise we prefer a separate line for the body of the method.

File: jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcMetadataLoader.java [hunk 3]
```diff
@@ -383,6 +343,52 @@ private void loadColumns(JdbcTable table, DatabaseMetaData metaData) {
...
+    private void processColumn(JdbcTable table, boolean convertLobs, ResultSet rs, int columnNumber) throws SQLException {
+        final String columnName = rs.getString(4);
+        if (_identifierQuoteString == null && new StringTokenizer(columnName).countTokens() > 1) {
+            logger.warn("column name contains whitespace: \"" + columnName + "\".");
+        }
+
+        final int jdbcType = rs.getInt(5);
+        final String nativeType = rs.getString(6);
+        final Integer columnSize = rs.getInt(7);
+        Integer decimalDigits = rs.getInt(9);
```
Comment (kaspersorensen) [inline jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcMetadataLoader.java:355]: We should add `final` to this one.

File: core/src/main/java/org/apache/metamodel/schema/ImmutableColumn.java [hunk 4]
```diff
@@ -71,12 +72,13 @@
      *            whether the column is a primary key or not
      */
     public ImmutableColumn(String name, ColumnType type, Table table, int columnNumber, Integer columnSize,
-            String nativeType, Boolean nullable, String remarks, boolean indexed, String quote, boolean primaryKey) {
+                           Integer decimalDigits, String nativeType, Boolean nullable, String remarks, boolean indexed, String quote, boolean primaryKey) {
```
Comment (arjansh) [inline core/src/main/java/org/apache/metamodel/schema/ImmutableColumn.java:75]: By adding a parameter to this constructor, the code will cause compilation errors in other project using this constructor. Can you please add an extra constructor with the signature of the original version of this constructor, which invokes this constructor with a default value for `decimalDigits`? In that manner you also don't have to change all the unit test code.

File: core/src/main/java/org/apache/metamodel/schema/MutableColumn.java [hunk 5]
```diff
@@ -70,9 +71,10 @@ public MutableColumn(String name, ColumnType type, Table table, int columnNumber
     }
 
     public MutableColumn(String name, ColumnType type, Table table, int columnNumber, Integer columnSize,
-            String nativeType, Boolean nullable, String remarks, boolean indexed, String quote) {
+                         Integer decimalDigits, String nativeType, Boolean nullable, String remarks, boolean indexed, String quote) {
```
Comment (arjansh) [inline core/src/main/java/org/apache/metamodel/schema/MutableColumn.java:74]: Same comment as for the `ImmutableColumn` constructor change, by adding a parameter to this constructor, the code will cause compilation errors in other project using this constructor. Can you please add an extra constructor with the signature of the original version of this constructor, which invokes this constructor with a default value for `decimalDigits`?

Comment (tkingsbury42) [reply core/src/main/java/org/apache/metamodel/schema/MutableColumn.java:85]: Fixed!

File: jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcMetadataLoader.java (hunk 3 above)
Comment (tkingsbury42) [reply jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcMetadataLoader.java:355]: Fixed!

File: core/src/main/java/org/apache/metamodel/schema/ImmutableColumn.java [hunk 6]
```diff
@@ -38,12 +38,55 @@
...
+     * @param indexed
+     *            whether the column is indexed or not
+     * @param quote
+     *            the quote character(s) of the column
+     * @param primaryKey
+     *            whether the column is a primary key or not
+     */
+    public ImmutableColumn(String name, ColumnType type, Table table, int columnNumber, Integer columnSize,
+                           String nativeType, Boolean nullable, String remarks, boolean indexed, String quote, boolean primaryKey) {
+        this.name = name;
```
Comment (arjansh) [inline core/src/main/java/org/apache/metamodel/schema/ImmutableColumn.java:76]: It is common practice to use constructor chaining when using multiple constructors for a java class, so then you could change this to:

```
        this(name, type, table, columnNumber, columnSize, null, nativeType, nullable, remarks, indexed, quote,
                primaryKey);
```

Comment (tkingsbury42) [reply core/src/main/java/org/apache/metamodel/schema/ImmutableColumn.java]: OK, this should be ready for another look.

File: core/src/main/java/org/apache/metamodel/schema/ImmutableColumn.java (hunk 4 above)
Comment (arjansh) [reply core/src/main/java/org/apache/metamodel/schema/ImmutableColumn.java:108]: This looks good now.

File: core/src/main/java/org/apache/metamodel/schema/MutableColumn.java (hunk 5 above)
Comment (arjansh) [reply core/src/main/java/org/apache/metamodel/schema/MutableColumn.java:85]: This looks good now.

File: core/src/main/java/org/apache/metamodel/schema/ImmutableColumn.java (hunk 6 above)
Comment (arjansh) [reply core/src/main/java/org/apache/metamodel/schema/ImmutableColumn.java]: This looks good now.

**Comment by arjansh** [comment]
All my comments have been addressed. @kaspersorensen Can you give it a second look, so that if you approve too, we can merge it?

File: jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcMetadataLoader.java (hunk 3 above)
Comment (kaspersorensen) [reply jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcMetadataLoader.java:355]: Doesn't look like it's fixed. Did you push the change?

**Comment by tkingsbury42** [comment]
That's a strange build failure. Is there a way to retrigger it and try again?

**Comment by arjansh** [comment]
@tkingsbury42 The easiest way to trigger the build is by pushing a new change. E.g. add one whitespace in a comment and then push the change.

File: jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcMetadataLoader.java (hunk 3 above)
Comment (tkingsbury42) [reply jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcMetadataLoader.java]: Doh. Pilot error. Look again!

**Pull request merged**